src/Elastic.CommonSchema/Base.Serialization.cs

```diff
@@ -96,16 +97,29 @@ protected internal virtual bool TryRead(string propertyName, out Type type)
 
 		public byte[] SerializeToUtf8Bytes() => JsonSerializer.SerializeToUtf8Bytes(this, GetType(), SerializerOptions);
 
+		[ThreadStatic]
+		private static ReusableUtf8JsonWriter _reusableJsonWriter;
+		private static ReusableUtf8JsonWriter ReusableJsonWriter => _reusableJsonWriter ??= new ReusableUtf8JsonWriter();
+
+		public StringBuilder Serialize(StringBuilder stringBuilder)
+		{
+			using var reusableWriter = ReusableJsonWriter.AllocateJsonWriter(stringBuilder);
+			reusableWriter.Serialize(this);
+			return stringBuilder;
```

Comment:
Any reason for having the StringBuilder as return parameter?

Reply:
Makes the method more composable e.g with linq's `.Aggregate()`